runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -147,7 +157,12 @@ public void write(IToken token, Object data) throws OverwriteException {
      * @return A result, which be cached.
      */
     public @Nonnull ILogData read(long address) {
-        if (!runtime.isCacheDisabled()) {
+        return read(address, null);
+    }
+
+    public @Nonnull ILogData read(long address, AddressSpaceOptions opts) {
+        if ((opts == null || opts != null && opts.isDoCache() )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)